### ISSUE TITLE
BUG: build failure with pip. list.remove(x): x not in list

### ISSUE THREAD

**Issue by oscarbenjamin** [scipy/scipy#16394]
### Describe your issue.

This comes from SymPy CI:
https://github.com/sympy/sympy/runs/6831736361?check_suite_focus=true

This is a followup after (https://github.com/scipy/scipy/issues/16308 and #16343). Now pip installing SciPy from git fails for a different reason giving the error message:
```
        File "/tmp/pip-build-env-_q1mz5xo/overlay/lib/python3.11/site-packages/mesonpy/__init__.py", line 430, in __init__
          self._metadata.dynamic.remove('version')
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      ValueError: list.remove(x): x not in list
      [end of output]
```

### Reproducing Code Example

```python
# Ubuntu 20.04.4 with Python 3.11b1 running in GA
sudo apt install antlr4 libgfortran5 gfortran libmpfr-dev libmpc-dev libopenblas-dev
pip install --upgrade pip wheel
pip install git+https://github.com/cython/cython@master
pip install git+https://github.com/numpy/numpy@main
pip install git+https://github.com/scipy/scipy@main
```

### Error message

```shell
Run pip install git+https://github.com/scipy/scipy@main
  pip install git+https://github.com/scipy/scipy@main
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.11.0-beta.3/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.0-beta.3/x64/lib
Collecting git+https://github.com/scipy/scipy@main
  Cloning https://github.com/scipy/scipy (to revision main) to /tmp/pip-req-build-8tim4693
  Running command git clone --filter=blob:none --quiet https://github.com/scipy/scipy /tmp/pip-req-build-8tim4693
  Resolved https://github.com/scipy/scipy to commit 329e162043bdb4d75bfecdbdbdad9ebcd9d60fe2
  Running command git submodule update --init --recursive -q
  Installing build dependencies: started
  Installing build dependencies: still running...
  Installing build dependencies: still running...
  Installing build dependencies: still running...
  Installing build dependencies: still running...
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [173 lines of output]
      The Meson build system
      Version: 0.62.2
      Source dir: /tmp/pip-req-build-8tim4693
      Build dir: /tmp/pip-req-build-8tim4693/.mesonpy-e3k1glay/build
      Build type: native build
      Project name: SciPy
      Project version: 1.10.0.dev0
      C compiler for the host machine: cc (gcc 9.4.0 "cc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0")
      C linker for the host machine: cc ld.bfd 2.34
      C++ compiler for the host machine: c++ (gcc 9.4.0 "c++ (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0")
      C++ linker for the host machine: c++ ld.bfd 2.34
      Host machine cpu family: x86_64
      Host machine cpu: x86_64
      Compiler for C supports arguments -Wno-unused-but-set-variable: YES
      Library m found: YES
      Fortran compiler for the host machine: gfortran (gcc 9.4.0 "GNU Fortran (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0")
      Fortran linker for the host machine: gfortran ld.bfd 2.34
      Program cython found: YES (/tmp/pip-build-env-_q1mz5xo/overlay/bin/cython)
      Program pythran found: YES (/tmp/pip-build-env-_q1mz5xo/overlay/bin/pythran)
      Program cp found: YES (/usr/bin/cp)
      Program python3 found: YES (/opt/hostedtoolcache/Python/3.11.0-beta.3/x64/bin/python)
      Found pkg-config: /usr/bin/pkg-config (0.29.1)
      Library npymath found: YES
      Library npyrandom found: YES
      Run-time dependency openblas found: YES 0.3.8
      Dependency openblas found: YES 0.3.8 (cached)
      Program _build_utils/cythoner.py found: YES (/opt/hostedtoolcache/Python/3.11.0-beta.3/x64/bin/python /tmp/pip-req-build-8tim4693/scipy/_build_utils/cythoner.py)
      Checking for function "open_memstream" : NO
      Configuring messagestream_config.h using configuration
      Checking for size of "void*" : 8
      Run-time dependency threads found: YES
      Checking for size of "void*" : 8
      Dependency threads found: YES unknown (cached)
      Compiler for C supports arguments -Wno-unused-but-set-variable: YES (cached)
      Compiler for C supports arguments -Wno-unused-but-set-variable: YES (cached)
      Compiler for C++ supports arguments -Wno-format-truncation: YES
      Compiler for C++ supports arguments -Wno-class-memaccess: YES
      Build targets in project: 201
      SciPy 1.10.0.dev0
        User defined options
          Native files: /tmp/pip-req-build-8tim4693/.mesonpy-native-file.ini
          debug       : false
          optimization: 2
          prefix      : /opt/hostedtoolcache/Python/3.11.0-beta.3/x64
          strip       : true
      Found ninja-1.10.2.git.kitware.jobserver-1 at /tmp/pip-build-env-_q1mz5xo/overlay/bin/ninja
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_ufuncs_pxi_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_ufuncs_pxi_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_ufuncs_pxi_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_ufuncs_pxi_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_ufuncs_pxi_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_stats_pxd' has more than one output! Using the first one.
      WARNING: custom_target 'cython_special' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_stats_gen_pyx' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_spatial_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_spatial_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target 'cython_linalg' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_lib_pxd' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_dummy_init_cyoptimize' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      WARNING: custom_target '_cython_tree' has more than one output! Using the first one.
      + meson setup --native-file=/tmp/pip-req-build-8tim4693/.mesonpy-native-file.ini -Ddebug=false -Dstrip=true -Doptimization=2 --prefix=/opt/hostedtoolcache/Python/3.11.0-beta.3/x64 /tmp/pip-req-build-8tim4693 /tmp/pip-req-build-8tim4693/.mesonpy-e3k1glay/build
      Traceback (most recent call last):
        File "/opt/hostedtoolcache/Python/3.11.0-beta.3/x64/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
          ^^^^^^
        File "/opt/hostedtoolcache/Python/3.11.0-beta.3/x64/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/hostedtoolcache/Python/3.11.0-beta.3/x64/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 130, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-_q1mz5xo/overlay/lib/python3.11/site-packages/mesonpy/__init__.py", line 805, in get_requires_for_build_wheel
          with _project(config_settings) as project:
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/hostedtoolcache/Python/3.11.0-beta.3/x64/lib/python3.11/contextlib.py", line 137, in __enter__
          return next(self.gen)
                 ^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-_q1mz5xo/overlay/lib/python3.11/site-packages/mesonpy/__init__.py", line 784, in _project
          with Project.with_temp_working_dir(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/hostedtoolcache/Python/3.11.0-beta.3/x64/lib/python3.11/contextlib.py", line 137, in __enter__
          return next(self.gen)
                 ^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-_q1mz5xo/overlay/lib/python3.11/site-packages/mesonpy/__init__.py", line 479, in with_temp_working_dir
          yield cls(source_dir, tmpdir, build_dir)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-_q1mz5xo/overlay/lib/python3.11/site-packages/mesonpy/__init__.py", line 430, in __init__
          self._metadata.dynamic.remove('version')
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      ValueError: list.remove(x): x not in list
      [end of output]
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.
note: This error originates from a subprocess, and is likely not a problem with pip.
Error: Process completed with exit code 1.
```

### SciPy/NumPy/Python version information

scipy/numpy/cython from git and python 3.11b3

**Comment by tylerjereddy** [comment]
Sounds like this issue: https://github.com/scipy/scipy/pull/16353#issuecomment-1151711888

Should be fixed when a new `meson-python` gets released I think (when combined with latest `pep621` package).

**Comment by rgommers** [comment]
Yes, this is fixed in `meson-python` 0.6.0, closing. Thanks for the report @oscarbenjamin
